src/main/java/org/mskcc/cbio/portal/scripts/GeneticAlterationIncrementalImporter.java

```diff
@@ -0,0 +1,114 @@
+
+    public GeneticAlterationIncrementalImporter(
+            int geneticProfileId,
+            List<Integer> fileOrderedSampleList
+    ) {
+
+        this.geneticProfileId = geneticProfileId;
+        this.fileOrderedSampleList = fileOrderedSampleList;
+    }
+
+    @Override
+    public boolean store(String[] values, CanonicalGene gene, String geneSymbol) throws DaoException {
+        int geneticEntityId = gene.getGeneticEntityId();
+        String[] expandedValues = extendValues(geneticEntityId, values);
+        return super.store(expandedValues, gene, geneSymbol);
+    }
+
+    @Override
+    public boolean store(int geneticEntityId, String[] values) throws DaoException {
+        String[] expandedValues = extendValues(geneticEntityId, values);
+        return super.store(geneticEntityId, expandedValues);
+    }
+
+    @Override
+    public void initialise() {
+        try {
+            this.geneticAlterationMap = daoGeneticAlteration.getGeneticAlterationMapForEntityIds(geneticProfileId, null);
+            ArrayList <Integer> savedOrderedSampleList = DaoGeneticProfileSamples.getOrderedSampleList(this.geneticProfileId);
+            int initialOrderSampleListSize = savedOrderedSampleList.size();
+            geneticAlterationMap.forEach((geneticEntityId, sampleToValue) -> {
+                if (sampleToValue.size() != initialOrderSampleListSize) {
+                    throw new IllegalStateException("Number of samples ("
+                            + sampleToValue.size() + ") for genetic entity with id "
+                            + geneticEntityId + " does not match with the number in the preexisting sample list ("
+                            + initialOrderSampleListSize + ").");
+                }
+            });
+            // add all new sample ids at the end
+            this.orderedSampleList = new ArrayList<>(savedOrderedSampleList);
+            Set<Integer> savedSampleSet = new HashSet<>(savedOrderedSampleList);
+            List<Integer> newSampleIds = this.fileOrderedSampleList.stream().filter(sampleId -> !savedSampleSet.contains(sampleId)).toList();
+            this.orderedSampleList.addAll(newSampleIds);
+            DaoGeneticProfileSamples.deleteAllSamplesInGeneticProfile(this.geneticProfileId);
+            daoGeneticAlteration.deleteAllRecordsInGeneticProfile(this.geneticProfileId);
+        } catch (DaoException e) {
+            throw new RuntimeException(e);
+        }
+        super.initialise();
+    }
+
+    @Override
+    public void finalise() {
+        expandRemainingGeneticEntityTabDelimitedRowsWithBlankValue();
+        super.finalise();
+    }
+
+    private String[] extendValues(int geneticEntityId, String[] values) {
+        Map<Integer, String> sampleIdToValue = mapWithFileOrderedSampleList(values);
+        String[] updatedSampleValues = new String[orderedSampleList.size()];
+        for (int i = 0; i < orderedSampleList.size(); i++) {
+            updatedSampleValues[i] = "";
```

Comment:
I am not sure what action points are here

Reply:
I guess my understanding of this topic (what kinds of metrics are acceptable in the genetic_alteration table) is simply inadequate. I think it has been expanded beyond the CNA event calls, so I would not feel comfortable making any decisions about how best to validate or import into that table. There is some guidance in the file format documentation about unset values, but I don't think it is clear enough. There also seems (to me) to be two places where we define the set of samples in a profile (at least for CNA profiles) ... we have the list of columns across the top headers of the file which give the sample list, and we also have case_lists/cases_cna.txt. I'm not sure which one should "rule" when it comes to data import .. or what should be done if there is a clash. I think it would be better to consolidate the definition of the sample list into one location .. or to require that both lists match exactly if that is not feasible. But I don't have any action points for this PR I guess.